src/odinapi/utils/smrl2filewriter.py

```diff
@@ -0,0 +1,215 @@
+            )
+            for scan in currentscans:
+                scans.append(scan["ScanID"])
+            start += relativedelta(days=1)
+        return scans
+
+
+@attr.s
+class L2FileCreater:
+    project = attr.ib(type=str)
+    freqmode = attr.ib(type=int)
+    product = attr.ib(type=str)
+    parameters = attr.ib(type=datamodel.L2File)
+    data = attr.ib(type=List[datamodel.L2Full])
+    outdir = attr.ib(type=str)
+
+    @property
+    def start(self) -> dt.datetime:
+        return self.data[0].l2.Time
+
+    @property
+    def end(self) -> dt.datetime:
+        return self.data[-1].l2.Time
+
+    @property
+    def ntimes(self) -> int:
+        return len(self.data)
+
+    @property
+    def nlevels(self) -> int:
+        return len(self.data[0].l2.Profile)
+
+    @property
+    def invmode(self) -> str:
+        return self.data[0].l2.InvMode
+
+    def filename(self) -> str:
+        if not os.path.isdir(self.outdir):
+            os.makedirs(self.outdir)
+        return os.path.join(
+            self.outdir,
+            datamodel.generate_filename(
+                self.project, self.product, self.start
+            )
+        )
+
+    @property
+    def header(self) -> Dict[str, str]:
+        return datamodel.get_file_header_data(
+            self.freqmode, self.invmode, self.product, self.start, self.end
+        )
+
+    def write_to_file(self) -> None:
+        with Dataset(self.filename(), "w", fomat="NETCFD4") as ds:
+            ds.createDimension("time", self.ntimes)
+            ds.createDimension('level', self.nlevels)
+            for para in self.header:
+                setattr(ds, para, self.header[para])
+            for para in self.parameters:
+                nc_var = ds.createVariable(
+                    para.name,
+                    para.dtype.value,
+                    para.dimension.value,
+                    zlib=True
+                )
+                nc_var.description = para.get_description(self.product)
+                nc_var.units = para.get_units(self.product).value
+                if para.units == datamodel.Units.time:
+                    nc_var[:] = date2num(
+                        [d.get_data(para) for d in self.data],
+                        para.units.value,
+                        calendar='standard'
+                    )
+                else:
+                    nc_var[:] = [d.get_data(para) for d in self.data]
+
+
+def get_l2data(
+    l2getter: L2Getter, start: dt.datetime, end: dt.datetime
+) -> List[datamodel.L2Full]:
+    scanids = l2getter.get_scanids(start, start + relativedelta(months=1))
+    return l2getter.get_data(scanids)
+
+
+def process_period(
+    db1: DatabaseConnector,
+    db2: level2db.Level2DB,
+    project: str,
+    freqmode: int,
+    product: str,
+    start: dt.datetime,
+    end: dt.datetime,
+    parameters: datamodel.L2File,
+    outdir: str
+) -> None:
+    l2getter = L2Getter(freqmode, product, db1, db2)
+    while start < end:
```

Comment:
periods are meant to be inclusive start to exclusive end?

Reply:
yes.